File: src/org/foonugget/kdict/TranslationsActivity.java

package org.foonugget.kdict;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.ListView;

import com.google.inject.Inject;

import org.foonugget.kdict.WordSearchListAdapter.Factory;
import org.foonugget.kdict.data.Dictionary;
import org.foonugget.kdict.data.WordMatch;

import roboguice.activity.RoboActivity;
import roboguice.inject.InjectResource;
import roboguice.inject.InjectView;

import java.util.ArrayList;
import java.util.List;

public class TranslationsActivity extends RoboActivity {

    private final static String TAG = TranslationsActivity.class
            .getSimpleName();

    @InjectView(R.id.translationsList)
    private ListView mtranslationsListView;

    @InjectResource(R.string.intentVarWord)
    private String WORD_KEY;

    @Inject
    Dictionary dict;

    @Inject
    private Factory wordSearchListAdapterFactory;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.translations);

        Intent intent = getIntent();
        String wordString = intent.getStringExtra(WORD_KEY);
        List<WordMatch> matchesList = dict.wordSearch(wordString);
        WordMatch match = matchesList.get(0);

        // String matches = match.getTranslation();
        String[] translations = match.getTranslation().split(",");
        List<WordMatch> allWords = new ArrayList<WordMatch>();
        for (String word : translations) {
            List<WordMatch> englishMatches = dict.findEnglishByKoreanExact(word);
            allWords.addAll(englishMatches);
            Log.i(TAG, englishMatches.toString());
        }

        WordSearchListAdapter adapter = wordSearchListAdapterFactory.create(matchesList);
        mtranslationsListView.setAdapter(adapter);
    }

}

File: app/src/main/java/org/foonugget/kdict/ui/TranslationsActivity.kt
package org.foonugget.kdict.ui

import android.os.Bundle
import android.util.Log
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import org.foonugget.kdict.R
import org.foonugget.kdict.data.Dictionary
import org.foonugget.kdict.data.WordMatch
import org.koin.android.ext.android.inject

class TranslationsActivity : AppCompatActivity() {

    private val dict: Dictionary by inject()
    private val wordSearchListAdapterFactory: WordSearchListAdapter.Factory by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.translations)

        val translationsListView = findViewById<ListView>(R.id.translationsList)
        val wordString = intent.getStringExtra(getString(R.string.intentVarWord)) ?: ""
        val matchesList = dict.wordSearch(wordString)
        if (matchesList.isEmpty()) {
            return
        }
        val match = matchesList[0]

        val translations = match.translation.split(",")
        val allWords = mutableListOf<WordMatch>()
        for (word in translations) {
            val englishMatches = dict.findEnglishByKoreanExact(word.trim())
            allWords.addAll(englishMatches)
            Log.i(TAG, englishMatches.toString())
        }

        val adapter = wordSearchListAdapterFactory.create(matchesList)
        translationsListView.adapter = adapter
    }

    companion object {
        private val TAG = TranslationsActivity::class.java.simpleName
    }
}
